### DISCUSSION TITLE
Should there be a way to execute tests multiple times in parallel?

### DISCUSSION THREAD

**Discussion by djmcgreal-cc** [bats-core/bats-core#730] [Ideas]
Hi. We would like to test race conditions. Is there already a way to ask Bats to run `@test`s multiple times in parallel? Frameworks that support this do so so that the test case code is kept free of the orchestration code.

At the moment we're running the whole suite in parallel but since the associations are at the `@test` level it makes it less likely that the relevant tests will be executed together.

**Comment by martin-schulze-vireso** [comment]
> Hi. We would like to test race conditions. Is there already a way to ask Bats to run `@test`s multiple times in parallel? Frameworks that support this do so so that the test case code is kept free of the orchestration code.

You mean several instances of the same test? In that case no. But it would be trivial to  run the same code from multiple tests and parallelize these.

> At the moment we're running the whole suite in parallel but since the associations are at the `@test` level it makes it less likely that the relevant tests will be executed together.

You can try to call single files and parallelize them. Else I would suggest to do the actual parallelization within the test.
